Short-Answer/Algorithms_Answers.md

```diff
@@ -1 +1,55 @@
+
+So if we fire this up with an n of 10 or so (I'm barely doing the math at this point, yes, because I don't care about the exact values now that I've seen the pattern), we have about 5 iterations of i (scales directly in relation to n), each of which brings about 5, then 4, then 3, then 2, then 1 iterations of j, as j loops over a smaller amount each time (an n log^n scale), each of which does something similar with k (another n log^n scale), and then brings another 4 or 5 iterations of l, just as a gotcha.
+
+
+This is something nightmarish. It explodes. I am not sure it is n^factorial, but it is O 2^n at least, and it's at least trying to make n^factorial feel nervous. I'll get a more exact number if I finish robot sort early.
+
+```
+c)  def bunnyEars(bunnies):
+      if bunnies == 0:
+        return 0 # Finish condition.
+      return 2 + bunnyEars(bunnies-1) # recurse once per bunny
+```
+So, if bunnies is 0, we're done, but if bunnies is 1, or 2, or 3, we're pushing a stack frame per bunny. But we aren't doing anything more complex than that. This algorithm will only have one iteration per bunny, so we might as well use a `for` loop, as that will save us from stack overflow, but like a typical single `for` loop, it's _just_ O(n) time once you simplify out that constant (using an iterator vs. using the stack).
+
+## Exercise II
+
+Suppose that you have an _n_-story building and plenty of eggs. Suppose also that an egg gets broken if it is thrown off floor _f_ or higher, and doesn't get broken if dropped off a floor less than floor _f_. Devise a strategy to determine the value of _f_ such that the number of dropped eggs is minimized.
+
+
+When n is 2 and f is 2, I have to throw 2 eggs to determine the value of f, But if n is 2 and f is 1 or 0 (effectively identical), I could find out f is 1 by throwing off the first floor first, and I would not need to throw another egg. So I should probably always test the first floor as my first attempt if I don't have a guarantee that f is non-zero, as that creates a boundary of assuring whether f is a positive whole number greater than 1 (one I can meaningfully search for). However, that doesn't resolve problems like when I am throwing off the Sears Tower (110 stories high!) or something similar, and f is some number like 50 (this seems implausible, but maybe I have very bouncy eggs, or this algorithm is secretly interacting with a very cushy trampoline).
+
+In order to find larger values of f amidst larger values of n with a minimal amount of egg tossing, I could throw an egg at floor n/2, call that floor t (for test), and then if it breaks, I would throw an egg at t/2, if it doesn't, I throw an egg at t+((n-t)/ 2), and so on, slicing the search range in half with each test.
```

Comment:
Thats right!  A binary search will be the best way to find our floor in question considering the goal is to minimize the number of drop trials.